fcli-other/fcli-functional-test/src/ftest/groovy/com/fortify/cli/ftest/fod/_common/FODAppRel.groovy

```diff
@@ -0,0 +1,84 @@
+/**
+ * Copyright 2023 Open Text.
+ *
+ * The only warranties for products and services of Open Text 
+ * and its affiliates and licensors ("Open Text") are as may 
+ * be set forth in the express warranty statements accompanying 
+ * such products and services. Nothing herein should be construed 
+ * as constituting an additional warranty. Open Text shall not be 
+ * liable for technical or editorial errors or omissions contained 
+ * herein. The information contained herein is subject to change 
+ * without notice.
+ */
+package com.fortify.cli.ftest.fod._common
+
+import com.fortify.cli.ftest._common.Fcli
+
+public class FODAppRel implements Closeable, AutoCloseable {
+    private final String random = System.currentTimeMillis()
+    private final String fcliVariableName = "fod_apprel_"+random
+    private final String appName = "fcli-"+random
+    private final String versionName = "v"+random
+    private final String ownerId = 16225;
+    
+    public FODAppRel createWebApp() {
+        Fcli.run("fod app create $appName:$versionName "+ 
+            "--description Auto\\ created\\ by\\ test " +
+            "--sdlc-status=Development " + 
+            "--release=$versionName "+
+            "--owner=$ownerId " +
+            "--app-type=Web " +
+            "--business-criticality=Medium " +
+            "--auto-required-attrs " +
+            "--store $fcliVariableName",
+            {it.expectSuccess(true, "Unable to create application release")})
+        return this
+    }
+    
+    public FODAppRel createMobileApp() {
+        Fcli.run("fod app create $appName:$versionName "+
+            "--description Auto\\ created\\ by\\ test " +
+            "--sdlc-status=Development " +
+            "--release=$versionName "+
+            "--owner=$ownerId " +
+            "--app-type=Mobile " +
+            "--business-criticality=Medium " +
+            "--auto-required-attrs " +
+            "--store $fcliVariableName",
+            {it.expectSuccess(true, "Unable to create application release")})
+        return this
+    }
+    
+    public FODAppRel createMicroservicesApp() {
+        Fcli.run("fod app create $appName:$versionName "+
+            "--description Auto\\ created\\ by\\ test " +
+            "--sdlc-status=Development " +
+            "--release=$appName:$versionName "+
```

Comment:
I'd suggest having a separate field for the microservice name, instead of using `$appName` as the microservice name. 

That's a potential reason for having static `create*` methods, allowing to have a random microservice name when creating microservice apps, while setting the `microserviceName` field to `null` for non-microservice apps. 

For even more flexibility, maybe we should employ the builder pattern, where each `fcli fod app create` option that may need to be customized (like app-type, and maybe app/ms/rel names) is represented by an instance field and corresponding builder method. For regular Java we have the Lombok `@Builder` annotation, I think Groovy may provide something similar.

Reply:
added an extra field for microservice name, if you want to rework it feel free

Reply:
Potentially, fcli could contain a bug where the application name is used instead of the specified microservice name when creating the microservice, hence it's better to have an explicit microservice name. Agree that for now it doesn't make sense to complicate things for having the `microserviceName` set to `null` when creating a non-microservices app.